app/logic/Mail/MailAccount.ts

```diff
@@ -118,13 +115,16 @@ export class MailAccount extends TCPAccount {
   }
 
   async save(): Promise<void> {
+    if (this.outgoing) {
+      this.outgoing.mainAccount = this;
+    }
+    console.log("mail account save", this.protocol, this.name, this, "outgoing", this.outgoing, "mainaccount", this.mainAccount, "storage", this.storage);
+    await super.save();
     await this.storage?.saveAccount(this);
```

Comment:
The problem here is that `super.save` tries to save all dependent accounts, which doesn't work until this account has been saved. This causes the creation of IMAP and POP3 accounts to fail, because they try to save the SMTP account first.